src/app/list/Survey.tsx

```diff
@@ -1,45 +1,56 @@
-import React, { useState } from 'react';
+import React from 'react';
 import { Choice } from '../../components/choice/Choice';
 import { Question } from '../../components/question/Question';
+import questionList from '../../questions.json';
 
 
-export function Survey(questions: Question[]) {
-
-  const questionEventHandler = (idx: number, selectedChoiceList: Choice[]) => {
-    questions[idx].choices = selectedChoiceList;
-  }
+export function Survey(/*questions: Question[]*/) {
+  const questions: Question[] = questionList;
 
   return (
     <div className="question-container">
       <ol className="question-form">
         {questions.map((question) => {
-          return QuestionUI(question, questionEventHandler)
+          return QuestionUI(question)
         })}
       </ol>
       <button className="button">Send</button>
     </div>
   )
 }
 
-function QuestionUI(question: Question,
-  questionEventHandler: (idx: number, selectedChoiceList: Choice[]) => void) {
+function QuestionUI(question: Question) {
+  let selectedChoices: Choice[] = [];
+
+  const selectChoice = (id: number) => {
+    if(!selectedChoices.includes(question.choices[id])) {
+      selectedChoices.push(question.choices[id]);
+    } else {
+      selectedChoices = selectedChoices.filter(choice => choice.id !== question.choices[id].id)
+      // it should have any notification like "you removed this choice"
+    }
+    console.log(selectedChoices);
+  };
 
   return (
-    <li className="question">
+    <li key={question.id} className="question">
       {question.question}
       <ul className="choice-container">
         {question.choices.map((choice) => {
-          return ChoiceUI(choice);
+          return ChoiceUI(choice, selectChoice);
         })}
       </ul>
     </li>
   );
 }
 
-function ChoiceUI(choice: Choice) {
+function ChoiceUI(
+  choice: Choice,
+  choiceHandler: (id: number) => void) {
+  
   return (
-    <li className="choice-list">
-      <button className="button choice">
+    <li key={choice.id} className="choice-list">
+      <button onClick={e => choiceHandler(choice.id)} className="button">
```

Comment:
I see that you are using button's for choices. Please check the implementation here.
https://github.com/bilginyuksel/survey-generator/commit/52220a43f8611b287f79f05acaa692436890f9ee
We should use inputs for this one and types as `checkbox` and `radio`. When you create radio inputs you can group them and you can manage this group very easily. And instead of listening the `onClick` you will listen the `onSelect` function.